src/routes/dashboard/(admin)/drafts/+page.server.ts

```diff
@@ -181,18 +185,27 @@ export const actions = {
     db.logger.info({ pairs }, 'intervening draft');
 
     await db.insertLotteryChoices(draftId, user.id, pairs);
-    for (const [studentUserId, labId] of pairs) {
-      const { name: labName } = await db.getLabById(labId);
-      const studentUser = await db.getUserById(studentUserId);
-      await dispatch.dispatchLotteryInterventionNotification(
-        labId,
-        labName,
-        studentUser.givenName,
-        studentUser.familyName,
-        studentUser.email,
-        draftId,
-      );
-    }
+
+    const jobs = (await Promise.all(
+      pairs.map(async ([studentUserId, labId]) => {
+        const [{ name: labName }, studentUser] = await Promise.all([
+          db.getLabById(labId),
+          db.getUserById(studentUserId),
+        ]);
+        return {
+          labId,
+          labName,
+          givenName: studentUser.givenName,
+          familyName: studentUser.familyName,
+          email: studentUser.email,
+          draftId,
+        };
+      }),
+    )) satisfies Parameters<
+      NotificationDispatcher['bulkDispatchLotteryInterventionNotification']
+    >[0];
```

Comment:
We can instead put the `satisfies` clause in the `return` expression. I think this is cleaner as it removes the mandatory parentheses.